tests/manage/cluster/test_must_gather.py

```diff
@@ -0,0 +1,180 @@
+import os
+import logging
+import time
+
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1, bugzilla
+from ocs_ci.ocs import openshift_ops, ocp
+from ocs_ci.ocs.utils import collect_ocs_logs
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@pytest.mark.polarion_id("OCS-1583")
+@bugzilla('1766646')
+class TestMustGather(ManageTest):
+
+    @pytest.fixture(autouse=True)
+    def init_ocp(self):
+        """
+        init OCP() object
+        """
+
+        self.ocs = openshift_ops.OCP()
+        self.ocp_obj = ocp.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+            if must_gather_pods:
+                for pod_to_del in must_gather_pods:
+                    self.ocp_obj.wait_for_delete(resource_name=pod_to_del)
+                    logger.info(f"deleted pods: {pod_to_del}")
+                    time.sleep(3)
+
+        request.addfinalizer(finalizer)
+
+    def test_must_gather(self):
+        """
+        Tests functionality of: oc adm must-gather
+
+        """
+
+        # Make logs root directory
+        logger.info("Creating logs Directory")
+        directory = self.make_directory()
+        logger.info(f"Creating {directory}_ocs_logs - Done!")
+
+        # Collect OCS logs
+        logger.info("Collecting Logs")
+        collect_ocs_logs(dir_name=directory, ocp=False)
+        logger.info("Collecting logs - Done!")
+
+        # Compare running pods list to "/pods" subdirectories
+        logger.info("Checking logs tree")
+        logs = self.get_log_directories(directory)
+        pods = self.get_ocs_pods()
+        logger.info(f"Logs: {logs}")
+        logger.info(f"pods list: {pods}")
+        assert set(sorted(logs)) == set(sorted(pods)), (
+            "List of openshift-storage pods are not equal to list of logs directories"
+            f"list of pods: {pods}"
+            f"list of log directories: {logs}"
+        )
+
+        # 2nd test: Verify logs file are not empty
+        logs_dir_list = self.search_log_files(directory)
+        assert self.check_file_size(logs_dir_list), (
+            "One or more log file are empty"
+        )
+
+    def make_directory(self):
+        """
+        Check if directory to store must gather logs already exist
+        and use new directory if so.
+
+        Returns:
+            str: Logs directory
+
+        """
+        index = 1
+        directory = f"/tmp/mg_test{index}"
```

Comment:
please import config and use config.RUN['log_dir']

Reply:
changed to  utilsocsci_log_path() instead - based on config.RUN['log_dir']